scripts/build-tools/settings.ts

```diff
@@ -0,0 +1,62 @@
+/* eslint-disable no-console */
+import fs from 'fs';
+
+import ini from 'ini';
+
+const OTT_GLOBAL_PLAYER_ID = 'M4qoGvUk';
+
+export const initSettings = (mode: string) => {
+  const localFile = `ini/.webapp.${mode}.ini`;
+  const templateFile = `ini/templates/.webapp.${mode}.ini`;
+
+  // The build ONLY uses .ini files in /ini to include in the build output.
+  // All .ini files in the directory are git ignored to customer specific values out of source control.
+  // However, this script will automatically create a .ini file for the current mode if it doesn't exist
+  // by copying the corresponding mode file from the ini/templates directory.
+  if (!fs.existsSync(localFile) && fs.existsSync(templateFile)) {
+    fs.copyFileSync(templateFile, localFile);
+  }
+
+  let settings: Record<string, string | undefined>;
+
+  try {
+    const data = fs.readFileSync(`ini/.webapp.${mode}.ini`, 'utf8');
+    settings = ini.parse(data);
+  } catch (err: unknown) {
+    settings = {};
+  }
+  const { defaultConfigSource, additionalAllowedConfigSources, playerId, playerLicenseKey, UNSAFE_allowAnyConfigSource } = settings;
+
+  // The ini file values will be used if provided, even if compile-time values are set
```

Comment:
The reason for the ini file was to explicitly not set these values at compile time, but rather to leave a mechanism for customers to adjust settings without needing to recompile.